biz.aQute.bndlib/src/aQute/bnd/header/Attrs.java

```diff
@@ -84,7 +86,29 @@ private Attrs(Map<String, String> map, Map<String, Type> types) {
 	}
 
 	public Attrs() {
-		this(new LinkedHashMap<>(), new HashMap<>());
+		this(createSortedMap(), new HashMap<>());
+	}
+
+	/**
+	 * Create a TreeMap with custom comparator that ensures consistent ordering:
+	 * - Attributes (no ':' suffix) come first, sorted alphabetically
+	 * - Directives (with ':' suffix) come second, sorted alphabetically
+	 */
+	private static Map<String, String> createSortedMap() {
+		Comparator<String> keyComparator = (k1, k2) -> {
+			boolean isDirective1 = k1.endsWith(":");
```

Comment:
Small suggestion: there are static helpers `Attrs.isAttribute(String)` and `Attrs.isDirective(String)` . Just for centralizing those specific checks.